### PR TITLE
feat: introduce recurrence form component for scheduling

### PR DESCRIPTION
**Pull request by scavnickyj**
JIRA: F1-437
risk: low

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

### CODE REVIEW

File: libs/sdk-ui-kit/src/RecurrenceForm/RepeatTypeSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        const isLastOccurenceOfWeekDay = isLastOccurrenceOfWeekdayInMonth(startDate);
+        return Object.values(RECURRENCE_TYPES).map((id): IDropdownItem => {
+            const localizationKey = getLocalizationKey(id);
+
+            return {
+                id,
+                title: intl.formatMessage(localizationKey, {
+                    day: getIntlDayName(intl, startDate),
+                    // week values >= 5 are translated as "last", but some months do not have "fifth week day"
+                    week: isLastOccurenceOfWeekDay ? 999 : getWeekNumber(startDate),
```
Comment (kandl) [inline libs/sdk-ui-kit/src/RecurrenceForm/RepeatTypeSelect.tsx:59]: is 999 correct here? Should it then be different intl key?

Comment (scavnickyj) [reply libs/sdk-ui-kit/src/RecurrenceForm/RepeatTypeSelect.tsx:59]: it works like this `"value": "Monthly on the {week, select, 1 {first} 2 {second} 3 {third} 4 {fourth} other {last}} {day}"` so should be ok

Comment (scavnickyj) [reply libs/sdk-ui-kit/src/RecurrenceForm/RepeatTypeSelect.tsx:59]: Redone in #5057

**Comment by scavnickyj** [comment]
Closing in favor of #5057

**Pull request closed without merging**